antipattern/processing-request-data-manually.md

```
@@ -0,0 +1,98 @@
+    if request.method == 'POST':
+        form = RegisterForm(request.POST, request.FILES)
+    # ...
+```
+
+We don't have to work with `form` to *render* it. If the `RegisterForm` has as fields an `email` field, a `password` and `password2` and a `read_terms` checkbox, then the `RegisterForm` can work with the request data submitted by a manually constructed HTML form, but regardless whether we render the form, or construct one manually, the form automates a lot of work we discussed above.
+
+A `Form` itself can first read the data from the `request.POST` and `request.FILES`, then validate that data, and eventually offers the cleaned data. If it is a `ModelForm`, it has even logic to create or update a record, regardless from where the data originates. The template only has to make sure that the name of the form elements is according to the form fields of the `Form`.
+
+We can also pass the `form` to the context, not to render it per se, but to inspects the `form.errors`, and show the errors on the fields accordingly.
+
+# Extra tips
```

markdownlint-cli2 (0.17.2)

Heading style
Expected: setext; Actual: atx

(MD003, heading-style)